src/Stryker.Utilities/Buildalyzer/IAnalyzerResultExtensions.cs

```diff
@@ -259,6 +260,7 @@ private static IEnumerable<string> ParseDiagnostics(string diagnostics)
 
         return diagnostics
             .Split(";")
+            .Select(x => Regex.Replace(x, "[^a-zA-Z0-9_.]+", "", RegexOptions.Compiled))
```

Comment:
Would stripping of newline and whitespace characters not be sufficient? I personally try to avoid regex whenever possible. This currently seems to strip any non-alphanumeric character and I think we are doing too much there. non-alphanumeric characters except newlines and `;` are invalid characters and we should not be more forgiving in that regard that msbuild itself imo. Let me know if I'm missing something here.

Reply:
Fair point, modified the clause accordingly.